### ISSUE TITLE
Parallel replicas: works slower with skip index

### ISSUE THREAD

**Issue by devcrafter** [ClickHouse/ClickHouse#54304]
https://pastila.nl/?00392915/f1bda54223840bf771c1f7d3315e3a5a#oDykBnqGhHc7BmSVteofsA==

**Comment by nickitat** [comment]
wouldn't it be also the case for single-node execution too? so what do we want to fix here exactly?
the only point I see here to address is to not do index analysis multiple times, because it might be quite costly

**Comment by devcrafter** [comment]
> wouldn't it be also the case for single-node execution too? so what do we want to fix here exactly? the only point I see here to address is to not do index analysis multiple times, because it might be quite costly

Yes, the first step towards this is to reuse snapshot and the result of analysis on initiator as parallel replicas coordinator working set

**Comment by devcrafter** [comment]
Done regarding this issue
- use min-max indices first - https://github.com/ClickHouse/ClickHouse/pull/60675
- with local plan for PR, there is no double index analysis on initiator anymore - https://github.com/ClickHouse/ClickHouse/pull/64448

**Comment by nickitat** [comment]
Do you think we should keep it open still?
